services/endpointsrv/service.go

```diff
@@ -438,17 +438,71 @@ func (e *EndpointService) EndpointCheckStatus(ctx context.Context, checkID strin
 
 	out := make([]types.EndpointCheckStep, len(steps))
 	for idx, step := range steps {
+		status, conclusion := StepStatusAndConclusion(step)
 		out[idx] = types.EndpointCheckStep{
-			StepID:    step.ID,
-			EvalID:    step.EvalID,
-			Input:     []byte(step.Input.String),
-			Output:    []byte(step.Output.String),
-			Assertion: step.Assertion.String,
+			StepID:     step.ID,
+			EvalID:     step.EvalID,
+			Input:      []byte(step.Input.String),
+			Output:     []byte(step.Output.String),
+			Assertion:  step.Assertion.String,
+			Status:     status,
+			Conclusion: conclusion,
 		}
 	}
 
+	status, conclusion := CheckStatusAndConclusion(out)
 	return types.EndpointCheck{
-		CheckID: checkID,
-		Steps:   out,
+		CheckID:    checkID,
+		Steps:      out,
+		Status:     status,
+		Conclusion: conclusion,
 	}, nil
 }
+
+// StepStatusAndConclusion infers the status of a step based on the contents of the database.
+func StepStatusAndConclusion(step db.UnweaveEndpointCheckStep) (types.CheckStatus, *types.CheckConclusion) {
```

Comment:
suggestion: these probably shouldn't be exported, especially if they are using the underlying database types.

Reply:
Also, might be worth having a test for this specific logic; just because there are a few `if`s in here.